server/pypi/packages/scipy/patches/chaquopy.patch

```diff
@@ -325,77 +244,34 @@ diff -aur src-orig/scipy/special/utils/makenpz.py src/scipy/special/utils/makenp
      '''
--    import numpy as np
--    ver = tuple(int(num) for num in np.__version__.split('.')[:2])
--    return ver < (1, 19)
-+    # Chaquopy: we force Numpy to 1.20 version so not old
-+    return False
-+    
- 
- 
- def make_biasedurn():
++    # Chaquopy: prevent the compiled parts of NumPy from being imported
++    import builtins
++    builtins.__NUMPY_SETUP__ = True
++
+     import numpy as np
+     ver = tuple(int(num) for num in np.__version__.split('.')[:2])
+     return ver < (1, 19)
 diff -aur src-orig/setup.py src/setup.py
 --- src-orig/setup.py	2023-09-20 11:08:03.732957618 +0200
 +++ src/setup.py	2023-09-21 13:13:08.707895078 +0200
-@@ -32,6 +32,17 @@
+@@ -32,6 +32,15 @@
  
  import builtins
  
-+# Chaquopy: an OpenBLAS test executable is built without using LDFLAGS, so we have to add
-+# --rpath-link here: see note in build-wheel.py.
++# Chaquopy
 +with open(os.path.join(os.path.dirname(__file__), "site.cfg"), "w") as site_file:
 +    site_file.write("[openblas]\n"
 +                    "library_dirs = {reqs_dir}/lib\n"
 +                    "include_dirs = {reqs_dir}/include\n"
-+                    "extra_link_args = -Wl,-rpath-link,{reqs_dir}/lib\n"
 +                    .format(reqs_dir=os.path.abspath("../requirements/chaquopy")))
 +
 +# Chaquopy: prevent the compiled parts NumPy from being imported
 +builtins.__NUMPY_SETUP__ = True
  
  CLASSIFIERS = """\
  Development Status :: 5 - Production/Stable
-@@ -230,23 +241,12 @@
-     Custom build_ext command to tweak extension building.
-     """
-     from numpy.distutils.command.build_ext import build_ext as npy_build_ext
-+
-+    # Chaquopy: force SCIPY_USE_PYTHRAN to 0 (not handled yet)
-     if int(os.environ.get('SCIPY_USE_PYTHRAN', 1)):
--        try:
--            import pythran
--            from pythran.dist import PythranBuildExt
--        except ImportError:
--            BaseBuildExt = npy_build_ext
--        else:
--            BaseBuildExt = PythranBuildExt[npy_build_ext]
--            _pep440 = importlib.import_module('scipy._lib._pep440')
--            if _pep440.parse(pythran.__version__) < _pep440.Version('0.10.0'):
--                raise RuntimeError("The installed `pythran` is too old, >= "
--                                   "0.10.0 is needed, {} detected. Please "
--                                   "upgrade Pythran, or use `export "
--                                   "SCIPY_USE_PYTHRAN=0`.".format(
--                                   pythran.__version__))
--    else:
--        BaseBuildExt = npy_build_ext
-+        os.environ['SCIPY_USE_PYTHRAN'] = "0"
-+    
-+    BaseBuildExt = npy_build_ext
- 
-     class build_ext(BaseBuildExt):
-         def finalize_options(self):
-@@ -492,8 +492,8 @@
-     run_build = parse_setuppy_commands()
-     if run_build:
-         try:
--            pkgname = 'numpy'
--            import numpy
-+            # pkgname = 'numpy'
-+            # import numpy
```

Comment:
Since this is only importing NumPy and not using it, it doesn't need to be removed.